### PR TITLE
Fix build vs. reparent race in resource.js

### PR DESCRIPTION
**Pull request by choumx**
Patch for #27483.

### CODE REVIEW

File: test/unit/test-resource.js [hunk 1]
```diff
@@ -103,6 +103,15 @@ describes.realWin('Resource', {amp: true}, (env) => {
     });
   });
 
+  it('should build if element is currently building', () => {
+    elementMock.expects('isBuilt').returns(false).once();
+    elementMock.expects('isBuilding').returns(true).once();
+    elementMock.expects('isUpgraded').returns(true).atLeast(1);
```
Comment (samouri) [inline test/unit/test-resource.js]: why `isUpgraded` more than once?

Comment (choumx) [reply test/unit/test-resource.js]: Copy pasta from above but I can change.

Comment (samouri) [reply test/unit/test-resource.js]: 🍝 delicious

**Pull request merged**